### ISSUE TITLE
RegExpMatch.groupNames throws if RegExp contains no named groups

### ISSUE THREAD

**Issue by leonsenft** [dart-lang/sdk#39406]
* Dart SDK Version: 2.6.0 in DartPad, 2.7.0-edge+google3-v2 (google3) on "linux_x64"
* Whether you are using Windows, MacOSX, or Linux (if applicable): Linux, macOS (haven't personally tried Windows, but this issue seems to appear on all platforms).
* Whether you are using Chrome, Safari, Firefox, Edge (if applicable): Chrome (for DartPad)

`RegExpMatch.groupNames` throws when invoked on a match produced by a `RegExp` with no named capture groups. This is reproducible on VM (Linux, macOS) and web (dart2js, DDC).

To reproduce, run the following:

```dart
void main() {
  final regExp = RegExp(r'\w+');
  final message = 'Hello world!';
  final match = regExp.firstMatch(message);
  print(match.groupNames);
}
```

In DartPad (dart2js):
```
Uncaught TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at _MatchImplementation.get$groupNames (<anonymous>:1970:42)
    at main (<anonymous>:1703:74)
    at dartMainRunner (<anonymous>:18:5)
    at <anonymous>:2469:7
    at <anonymous>:2455:7
    at dartProgram (<anonymous>:2466:5)
    at <anonymous>:2473:3
    at replaceJavaScript (frame_dark.html:38)
    at messageHandler (frame_dark.html:54)
```

In Chrome (DDC):
```
Uncaught TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at dart_library.library._js_helper._MatchImplementation.new.get groupNames [as groupNames] (dart_sdk.ddk.js?______buster=5302617.2019-11-15T16%3A24%3A34.000:20596)
    at Object.main$ [as main] (main.dart:5)
```

In VM:
```
The getter 'length' was called on null.
Receiver: null
Tried calling: length
dart:core              print
main.dart 5:3          main
```

This doesn't seem intended. I'd expect it to simply return an empty iterable when no named capture groups are defined. Note this getter *does* work when a named capture group is defined:

```dart
void main() {
  final regExp = RegExp(r'(?<name>\w+)');
  final message = 'Hello world!';
  final match = regExp.firstMatch(message);
  print(match.groupNames);
}
```

Produces:
```
(name)
```

_Note it appears that the returned iterable is backed by a `Set`._

**Comment by kevmoo** [comment]
@lrhn fixed in 9e10ef44587fd1dc419582c1b1501d08a81c29d1 ?

**Comment by lrhn** [comment]
I believe so, yes. Finally tracked down the last issues in JS compilation.
